theseus/optimizer/linear/lu_cuda_sparse_solver.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict, List, Optional, Type
+
+import torch
+
+from theseus.core import Objective
+from theseus.extlib.cusolver_lu_solver import CusolverLUSolver
+from theseus.optimizer import Linearization, SparseLinearization
+from theseus.optimizer.autograd import LUCudaSolveFunction
+
+from .linear_solver import LinearSolver
+
+
+class LUCudaSparseSolver(LinearSolver):
```

Comment:
We should move this to `linear/sparse_solver.py` so change that file to be Cholmod sparse solver since that's the only one in there currently. Same for the test file.

Reply:
I would prefer keeping files separate, it looks cleaner (even more so if the cuda class will have a mock version when cuda is not available). Anyway all is re-exported in the `__init__` file so it makes no difference to the user right?

Reply:
That sounds fine too. Then, after we decide what to do with the class hierarchy (from the other comment below) for the solver we should either rename sparse solver to cholmod sparse or make a new cholmod sparse with only the parent class in sparse solver to make things consistent.